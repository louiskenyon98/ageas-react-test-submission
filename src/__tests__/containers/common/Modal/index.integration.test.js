import React from 'react';
import {mount} from 'enzyme';
import thunk from 'redux-thunk';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import ModalContainer from '../../../../app/containers/common/Modal/index';

describe('ModalContainer Integration Test', () => {
    let props = {};
    let store = {};
    let mountedComponent;
    const middleware = [thunk];
    const mockStore = configureMockStore(middleware);
    let wrapper = () => {
        if (!mountedComponent) {
            mountedComponent = mount(
                <Provider store={store}>
                    <ModalContainer {...props} />
                </Provider>
            )
        }
        return mountedComponent
    };
    beforeEach(() => {
        props = {};
        store = mockStore({
            common: {
                modal: {
                    show: true,
                    type: "status",
                    props: {
                        status: "success",
                        title: "Note edited",
                        body: "note edited body"
                    }
                }
            }
        });
        mountedComponent = undefined;
    });
    it('should render ModalContainer component', () => {
        expect(wrapper()).toMatchSnapshot();
    })
});
