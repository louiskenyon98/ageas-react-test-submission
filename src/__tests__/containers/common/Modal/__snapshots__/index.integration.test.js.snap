// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContainer Integration Test should render ModalContainer component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ModalContainer)>
    <ModalContainer
      closeModal={[Function]}
      modal={
        Object {
          "props": Object {
            "body": "note edited body",
            "status": "success",
            "title": "Note edited",
          },
          "show": true,
          "type": "status",
        }
      }
    >
      <Modal
        close={[Function]}
        show={true}
      >
        <div
          className="veil"
          onClick={[Function]}
        />
        <div
          className="modal"
        >
          <StatusMessage
            body="note edited body"
            status="success"
            title="Note edited"
          >
            <h1>
              Note edited
            </h1>
            note edited body
          </StatusMessage>
        </div>
      </Modal>
    </ModalContainer>
  </Connect(ModalContainer)>
</Provider>
`;
